### ISSUE TITLE
Memory leaks in iOS

### ISSUE THREAD

**Issue by binkumar** [oblador/react-native-keychain#174]
When I'm using below code in my app.
```
Keychain.getGenericPassword()
      .then(credentials => {
        if (credentials) {
          this.setState({
            password: credentials.password
          });
        } else {
          return Keychain.setGenericPassword(
            'test',
            'randomSecret'
          ).then(() => {
            this.setState({
              password: 'test'
            });
            Promise.resolve('randomSecret')
          });
        }
      })
      .catch(error => {
        this.setState({
          password: error.message
        });
        Promise.reject(error)
      });
```

I'm getting memory leaks. I'm using Xcode instrumentation to identify this leaks. See attached screenshots.
<img width="1016" alt="screen shot 2018-12-10 at 1 26 39 pm" src="https://user-images.githubusercontent.com/6251880/49721220-c1fa7000-fc87-11e8-953c-af80a21aa225.png">
<img width="760" alt="screen shot 2018-12-10 at 1 26 50 pm" src="https://user-images.githubusercontent.com/6251880/49721221-c2930680-fc87-11e8-9163-83fbbbb9f971.png">
This leaks increases continuously until the app is an active. 

Below is the app leak area.
OSStatus osStatus = SecItemCopyMatching((__bridge CFDictionaryRef) query, (CFTypeRef*)&foundTypeRef);

Let me know whether I'm doing anything wrong.

**Comment by rajivshah3** [comment]
I'm seeing this as well. I found an [answer](https://stackoverflow.com/questions/16780202/secitemcopymatching-still-leak-on-osx-under-arc) on StackOverflow that seems to explain the issue. Here's an excerpt:
>You receive the keychain object from a CF-style function with Copy in its name. Therefore it has a +1 reference count, and you have the responsibility of explicitly releasing it when you're done using it. It is never released by your sample code, so it's leaking.

![image](https://user-images.githubusercontent.com/19519564/50569251-7c810e00-0d2e-11e9-8ffa-ccdfc64cdb27.png)

That could be the problem here, because Instruments is showing `Retain & Release Count: +1`. Once we're done with `osStatus`, maybe `CFRelease(osStatus)` will do the trick?

**Comment by Bautista-Baiocchi-lora** [comment]
Is there a fix for this?

**Comment by rajivshah3** [comment]
@Bautista-Baiocchi-lora when I have some time I'll try the `CFRelease(osStatus)` to see if it resolves the issue. I'm not very experienced with memory management though, so I'm not sure if this is the correct fix. @oblador do you know if this is the right approach?

**Comment by rajivshah3** [comment]
Update: The leak was actually coming from `foundTypeRef`, which was a Core Foundation object that we created but didn't release after we were done using it. I fixed it in #188

**Comment by binkumar** [comment]
@rajivshah3 Thanks for fixing the issue.

**Comment by Bautista-Baiocchi-lora** [comment]
> Update: The leak was actually coming from `foundTypeRef`, which is a Core Foundation object that we created but didn't release after we were done using it. I fixed it in #188

Great find! Thank you for the fix.
